### PR TITLE
Handle Hex String in EA Telemetry

### PR DESCRIPTION
**Pull request by cawthorne**
#DF-20580
The EA Telemetry currently fails for EAs where their result is a hex string:
https://smartcontract-it.atlassian.net/browse/DF-20580

https://chainlink-core.slack.com/archives/C0117GGJB6Y/p1729091653229629

Alternative implementation which edits ToDecimal directly, but effects other services is here:
https://github.com/smartcontractkit/chainlink/pull/14841

### CODE REVIEW

**Review by mxiao-cll — DISMISSED** [review]

**Review by george-dorin — CHANGES_REQUESTED** [review]
It seems that the behavior of not being able to convert hex to decimal is expected by Median and Mercury. As an alternative, perhaps we could add the hex conversion logic specifically for telemetry parsing?

**Review by george-dorin — DISMISSED** [review]

**Comment by cawthorne** [comment]
@george-dorin I managed to get all tests passing with the original approach with changing ToDecimal:
https://github.com/smartcontractkit/chainlink/pull/14841

My original patch had a bug in.

I was unsure if your comment `It seems that the behavior of not being able to convert hex to decimal is expected by Median and Mercury`, was based on the tests failing, or if that is actually the case.

We can choose which approach we want, and close the PR we don't merge in.

**Comment by george-dorin** [comment]
@cawthorne I was basing this on the tests, and it seems to me that `ToDecimal` should know how to interpret hex. However, I'm not entirely sure if other products are expecting the conversion to fail

**Comment by cawthorne** [comment]
@george-dorin I am fine with either, perhaps it would make sense to just do the telemetry at first, and then the more general solution could come later if there's a need.

File: core/services/ocrcommon/telemetry.go
Comment (pavel-raykov) [inline core/services/ocrcommon/telemetry.go:1]: Could you please update the comment of the function getJsonParsedValue?

Comment (cawthorne) [reply core/services/ocrcommon/telemetry.go:1]: Thanks, I added some extra comments about the issue. I can remove some detail from them if you think it is too much.

Comment (pavel-raykov) [reply core/services/ocrcommon/telemetry.go:1]: It is fine. Thanks.

**Review by pavel-raykov — DISMISSED** [review]

**Review by cedric-cordenier — DISMISSED** [review]

**Review by george-dorin — DISMISSED** [review]

**Review by pavel-raykov — APPROVED** [review]

**Review by george-dorin — APPROVED** [review]

**Pull request merged**